src/nft/hooks/useSendTransaction.ts

```diff
@@ -63,13 +64,15 @@ export const useSendTransaction = create<TxState>()(
               nftsPurchased,
               nftsNotPurchased,
               txReceipt,
+              purchasedWithErc20,
             }
           } else {
             set({ state: TxStateType.Failed })
             return {
               nftsPurchased: [],
               nftsNotPurchased: selectedAssets,
               txReceipt,
+              purchasedWithErc20,
```

Comment:
It doesn't seem like this property is ever actually ingested in an event. Additionally is there a way to parse that a tx used erc20 from the route response or receipt so this boolean doesn't have to be passed through 3 functions? Alternatively could we set `purchasedWithErc20` or some equivalent field in the `useSendTransaction` hook?